### ISSUE TITLE
Incompatible with node-sass: Import directives may not be used within control directives or mixins

### ISSUE THREAD

**Issue by papjul** [vinorodrigues/bootstrap-dark-5#12]
Has anyone succeed in implementing bootstrap-dark-5 in Webpack?

I added `includePaths = ["node_modules", "node_modules/bootstrap/scss"]` to the sass loader configuration as seen in #9 

However:
```
SassError: Import directives may not be used within control directives or mixins.
        on line 78 of node_modules/bootstrap-dark-5/scss/bootstrap-nightshade.scss
        from line 78 of node_modules/bootstrap-dark-5/scss/bootstrap-nightshade.scss, in mixin `@content`
        from line 15 of node_modules/bootstrap-dark-5/scss/dark/mixins/_color-scheme.scss, in mixin `prefers-color-scheme`
        from line 73 of node_modules/bootstrap-dark-5/scss/bootstrap-nightshade.scss
        from line 13 of assets/styles/app.scss
>>   @import "dark/root";
   ----------^
```

If it helps, I'm including from my custom scss file like this:
@import "../../node_modules/bootstrap-dark-5/scss/bootstrap-nightshade.scss";

Versions used:
* node-sass: 5.x 
* sass-loader: 11.x

**Comment by vinorodrigues** [comment]
Not sure how to help here.  I've not really built this for WebPack and I know zero on the subject.  I'll keep this open in case someone from the community can help.

**Comment by papjul** [comment]
Thank you, indeed I guessed you weren't using Webpack.

However, you may help as the issue itself says "SassError".

If I rephrase, it's saying you cannot have:
```
@import "dark/root";
```

inside:
```
@include prefers-color-scheme("html.dark") {
}
```

If it helps, I'm including from my custom scss file like this:
@import "../../node_modules/bootstrap-dark-5/scss/bootstrap-nightshade.scss";

Versions used:
* node-sass: 5.x 
* sass-loader: 11.x

**Comment by papjul** [comment]
OK, solved the issue by replacing SASS implementation from node-sass to (dart-)sass.

You are using non-standard features of the SASS language: https://github.com/sass/node-sass/issues/1471#issuecomment-213404643

I don't know if you plan on being compatible with the SASS language and I should leave this issue open?
In either case, could you add "Include bootstrap-dark-5 in an existing project" instructions?
Basically:
* Use (dart-)sass, and not node-sass
* Add `node_modules/bootstrap/scss` to the "load-path" (command line) / "includePath" (webpack rule)

**Comment by vinorodrigues** [comment]
Remember my code is just a rehash of Bootstrap 5 core code to add "dark mode" functionality.  If you look at Bootstrap 5 docs .. especially here (https://getbootstrap.com/docs/5.0/getting-started/build-tools/#sass) you'll note that they have deprecated the use of node-sass in favour of dart-sass.

They in turn cite this (https://sass-lang.com/blog/libsass-is-deprecated) article wherein the Sass curators state they will no longer support LibSass.

As such .. I don't think that this will change much.

**Comment by benbalter** [comment]
> In either case, could you add "Include bootstrap-dark-5 in an existing project" instructions?
Basically:
>
> Use (dart-)sass, and not node-sass
> Add node_modules/bootstrap/scss to the "load-path" (command line) / "includePath" (webpack rule)

👍🏻 to documenting these steps, even if this issue is ultimately a #wontfix... having this in the README would have saved me nearly an hour of debugging.

**Comment by vinorodrigues** [comment]
fair enough .. I'll add it to the README.

**Comment by benbalter** [comment]
@vinorodrigues Thanks for the README update and for clarifying the project status. Really appreciate the project. Once I figured out the dart issue, it was a drop-in replacement for my personal site. Thanks for creating this.
